utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala

```diff
@@ -0,0 +1,278 @@
+
+  @Documentation(description = "Returns the largest element (elements must be comparable)")
+  @GenericType(typingFunction = classOf[ListElementTyping])
+  def max[T <: Comparable[T]](@ParamName("list") list: java.util.Collection[T]): T = Collections.max[T](list)
+
+  @Documentation(description = "Returns a slice of the list starting with start index (inclusive) and ending at stop index (exclusive)")
+  @GenericType(typingFunction = classOf[ListTyping])
+  def slice[T](@ParamName("list") list: java.util.Collection[T], @ParamName("start") start: Int, @ParamName("stop") stop: Int): java.util.List[T] =
+    list.asScala.slice(start, stop).toList.asJava
+
+  private val numberPriorities = Map(
+    classOf[java.lang.Short] -> 1,
+    classOf[java.lang.Byte] -> 2,
+    classOf[java.lang.Integer] -> 3,
+    classOf[java.lang.Long] -> 4,
+    classOf[java.math.BigInteger] -> 5,
+    classOf[java.lang.Float] -> 6,
+    classOf[java.lang.Double] -> 7,
+    classOf[java.math.BigDecimal] -> 8,
+  ).asInstanceOf[Map[Class[_], Int]]
+
+  private def getNumberPriority(number: Number): Int = numberPriorities
+    .getOrElse(
+      number.getClass,
+      throw new IllegalArgumentException(s"Can't calculate the sum of elements. Unexpected element of type ${number.getClass.getName}: $number")
+    )
+
+  def toBigDecimal(number: Number): java.math.BigDecimal = {
+    number match {
+      case x: java.lang.Byte => java.math.BigDecimal.valueOf(x.intValue())
+      case x: java.lang.Short => java.math.BigDecimal.valueOf(x.intValue())
+      case x: Integer => java.math.BigDecimal.valueOf(x.intValue())
+      case x: java.lang.Long => java.math.BigDecimal.valueOf(x)
+      case x: java.lang.Float => java.math.BigDecimal.valueOf(x.doubleValue())
+      case x: java.lang.Double => java.math.BigDecimal.valueOf(x)
+      case x: java.math.BigDecimal => x
+      case x: java.math.BigInteger => new java.math.BigDecimal(x)
+      case x => throw new IllegalArgumentException(s"Can't convert to BigDecimal, unsupported number of type ${x.getClass.getName}: $x")
+    }
+  }
+
+  def toBigInteger(number: Number): java.math.BigInteger = {
+    number match {
+      case x: java.lang.Byte => java.math.BigInteger.valueOf(x.intValue())
+      case x: java.lang.Short => java.math.BigInteger.valueOf(x.intValue())
+      case x: Integer => java.math.BigInteger.valueOf(x.intValue())
+      case x: java.lang.Long => java.math.BigInteger.valueOf(x)
+      case x: java.math.BigInteger => x
+      case x => throw new IllegalArgumentException(s"Can't convert to BigInteger, unsupported number of type ${x.getClass.getName}: $x")
+    }
+  }
+
+  @Documentation(description = "Returns a sum of all elements")
+  @GenericType(typingFunction = classOf[ListElementTyping])
+  def sum[T <: java.lang.Number](@ParamName("listOfNumbers") list: java.util.Collection[T]): T = {
+    if (list.isEmpty) {
+      return 0.asInstanceOf[T]
+    }
+
+    val targetType = list.asScala.reduce { (a, b) =>
```

Comment:
1. We could do this check only when `typingFunction` return `Number`. For statically known element type, we can reuse `ListElementTyping` logic. Thanks to that we will be sure that type in runtime is the same as during compilation. Also we would avoid one cycle of checking the list
2. You have a bug here - please add tests for one element lists
3. IMO we should behave similarly to `NumberTypesPromotionStrategy.ForLargeNumbersOperation` - I mean list of bytes should be promoted to `Long`. Otherwise, it will be easy to overflow types, because lists can contain many elements. WDYT?

Reply:
Ad. 1. Let's add a TODO, that we should pass returnType to this method, so we could be sure that we return the same type as typed and info in which cases it won't work correctly (empty list with return type) or slower than necessary (statically known type)